azure-iot-device/tests/common/test_mqtt_transport.py

```diff
@@ -208,6 +294,32 @@ def test_calls_event_handler_callback(self, mocker, mock_mqtt_client, transport)
         assert callback.call_count == 1
         assert callback.call_args == mocker.call()
 
+    @pytest.mark.parametrize(
+        "error_params",
+        conack_return_codes,
+        ids=["{}->{}".format(x["name"], x["error"].__name__) for x in conack_return_codes],
+    )
+    @pytest.mark.it(
+        "Triggers on_mqtt_connection_failure event handler callback upon connect completion with failed RC"
```

Comment:
1) "event handler callback" is redundant. Just "event handler" should suffice.
2) "Triggers on_mqtt_connection_failure event handler with custom Exception...." perhaps? This makes the Exception (which will display as a result of parametrization) make more sense in the requirement.